templates/nginx/galaxy-main.j2

```diff
@@ -181,6 +181,16 @@ server {
 		proxy_set_header Connection "upgrade";
 	}
 
+	# Route all path-based interactive tool requests to the InteractiveTool proxy application
+    location ~* ^/(interactivetool/.+)$ {
+        proxy_redirect off;
+        proxy_http_version 1.1;
+        proxy_set_header Host $host;
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_set_header Upgrade $http_upgrade;
+        proxy_set_header Connection "upgrade";
+        proxy_pass http://127.0.0.1:{{ gie_proxy_port }};
```

Comment:
I have a question: I see on the main nginx conf that the path `/gie_proxy` is suffixed after the port [here](https://github.com/usegalaxy-eu/infrastructure-playbook/blob/master/templates/nginx/galaxy-main.j2#L171). Is that necessary for this `proxy_pass` as well?

Reply:
Actually, I do not know what the path

```
	# Global GIE configuration
	location /gie_proxy {
		proxy_pass http://127.0.0.1:8800/gie_proxy;
		proxy_redirect off;
	}
```

does (maybe @hexylena has a clue? <- git blame). As far as I know, the configuration that is making domain-based interactive tools work is [this one](https://github.com/usegalaxy-eu/infrastructure-playbook/blob/fce2056fbe7f06245dd47bb453a618c832de2602/templates/nginx/galaxy-ie-main.j2#L14-L26).

```
	server_name ~^(?<key>[0-9a-z-]*)\.ep\.interactivetool\.(?<domain>[a-z0-9.-]*)usegalaxy\.eu$;

	[...]

	location / {
		proxy_pass http://127.0.0.1:8800;
		proxy_redirect off;
		proxy_http_version 1.1;
		proxy_set_header Host $host;
		proxy_set_header Upgrade $http_upgrade;
		proxy_set_header Connection $connection_upgrade;
	}
```

See also the [Galaxy docs](https://docs.galaxyproject.org/en/latest/admin/special_topics/interactivetools.html#nginx-proxy-server-configuration-in-production). Both in the excerpt above and the Galaxy docs you can see that it is not necessary to add the prefix again. I assume it's needed only if you want special behavior (e.g. alter the prefix).

Reply:
True. Adding an extra prefix would OTOH mess up path-based ITs. 

Also the solution for supporting path-based ITs might differ from tool to tool. The config @sanjaysrikakulam refers to above will work if all paths in the web server in the IT are relative. If not, other solutions for injecting the path are available, as
documented in the Galaxy admin docs.